VibeSomeMauiBro/Views/CollectionPage.xaml.cs

```diff
@@ -0,0 +1,50 @@
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using VibeSomeMauiBro.Models;
+using VibeSomeMauiBro.Services;
+
+namespace VibeSomeMauiBro.Views;
+
+public partial class CollectionPage(ICatService catService) : ContentPage, INotifyPropertyChanged
+{
+    private readonly ICatService _catService = catService;
+    
+    private ObservableCollection<Cat> _likedCats = new();
+    public ObservableCollection<Cat> LikedCats 
+    { 
+        get => _likedCats;
+        set 
+        {
+            _likedCats = value;
+            OnPropertyChanged();
+        }
+    }
+
+    protected override void OnAppearing()
+    {
+        base.OnAppearing();
+        InitializeComponent();
+        BindingContext = this;
```

Comment:
Ok, my bad, don't use a primary constructor for this. We need to call these two in a constructor:
```csharp
InitializeComponent();
BindingContext = this;
```